kv/memberlist/memberlist_client_test.go

```diff
@@ -519,59 +526,163 @@ func TestMultipleCAS(t *testing.T) {
 	}
 }
 
+func defaultKVConfig(i int) KVConfig {
+	id := fmt.Sprintf("Member-%d", i)
+	var cfg KVConfig
+	flagext.DefaultValues(&cfg)
+	cfg.NodeName = id
+
+	cfg.GossipInterval = 100 * time.Millisecond
+	cfg.GossipNodes = 3
+	cfg.PushPullInterval = 5 * time.Second
+
+	cfg.TCPTransport = TCPTransportConfig{
+		BindAddrs: []string{"localhost"},
+		BindPort:  0, // randomize ports
+	}
+
+	return cfg
+}
+
 func TestMultipleClients(t *testing.T) {
-	c := dataCodec{}
+	members := 10
 
-	const members = 10
-	const key = "ring"
+	err := testMultipleClientsWithConfigGenerator(t, members, defaultKVConfig)
+	require.NoError(t, err)
+}
 
-	var clients []*Client
+func TestMultipleClientsWithMixedLabelsAndExpectFailure(t *testing.T) {
+	members := 3
```

Comment:
[Nit] Might as well be a constant:
```suggestion
	const members := 3
```